### PR TITLE
[fix] `get_all_block_hashes_by_height` returns empty hashmap if no blocks at height

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Issue https://github.com/near/nearcore/issues/8057

This PR changes the default behavior of `ChainStore::get_all_block_hashes_by_height` function to return an empty hashmap if no blocks exist for a given height.

### CODE REVIEW

**Comment by jakmeier** [comment]
@Longarithm do you want to review this, since you opened #8057?

**Review by Longarithm — CHANGES_REQUESTED** [review]
In the CI, `tests::client::process_blocks::produce_block_with_approvals` is failing. I can see that not all callers are modified, like in `chain/chain/src/chain.rs` and `chain/client/src/client.rs`.

One of ideas of the fix is to make sure that `StorageInternalError` is caught properly. So, logic like
```
if let Ok(epoch_id_to_blocks) = self.store.get_all_block_hashes_by_height(header.height()) {
  ...
}
```
should be changed to:
```
```
let epoch_id_to_blocks = self.store.get_all_block_hashes_by_height(header.height())?; // catches internal errors
if !epoch_id_to_blocks.is_empty() { // normal logic
  ...
}
```

Let me know if you can't find why test is failing, because it is seems quite complex to analyze.

**Comment by shreyan-gupta** [comment]
I couldn't see anything obviously wrong with the `produce_block_with_approvals` test. Infact, I believe that the test is probably flaky as on running it multiple times, I get a different number of block approvals (sometimes 6, sometimes 7) of the expected 8. Occasionally the test passes :confused:

File: chain/client/src/client.rs [hunk 1]
```diff
@@ -1748,7 +1748,7 @@ impl Client {
                                     approval,
                                     approval_type,
                                     true,
-                                    near_chain::Error::Other(format!(
+                                    near_chain::Error::DBNotFoundErr(format!(
```
Comment (Longarithm) [inline chain/client/src/client.rs:1751]: Oh, it is neat.

**Review by Longarithm — APPROVED** [review]
For the context - we discussed the test `produce_block_with_approvals` offline and were able to find that `handle_process_approval_error` needed to respect the new behaviour. Now tests pass, so congratulations with the first completed nearcore PR I guess :)

It's suboptimal that it was caught only by flaky and scary-looking integration test though. @robin-near is there a plan to migrate stuff like `produce_block_with_approvals` to TestLoop as well?

**Pull request merged**